homeassistant/components/mqtt/__init__.py

```diff
@@ -139,8 +138,8 @@ def valid_publish_topic(value):
 # Service call validation schema
 MQTT_PUBLISH_SCHEMA = vol.Schema({
```

Comment:
The Required above was copied from here. Are QOS and RETAIN required here also ? Or could it be set to optional ?